### PR TITLE
More character setup changes

### PR DESCRIPTION
**Pull request by SvartaSvansen**
* "Choose color" dialogs in character setup open with the current color as default (C.M.A. already does this).
* Random underwear/undershirt/socks generation now filters by allowed species.
* Added a check for allowed species for underwear/undershirt/socks when switching species in character setup; randomly generates a new one (or "Nude" if not applicable) if it doesn't pass.
* Added Vulpkanin to default species_allowed for underwear and undershirts (a rude oversight!).
* Moved redundant species_allowed lists in non-Vox-exclusive socks datums to the parent type.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
@SvartaSvansen 


Ready?

**Comment by SvartaSvansen** [comment]
Ready for what?

**Comment by ZomgPonies** [comment]
Merging

**Comment by SvartaSvansen** [comment]
Oh. Yes, it is.

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -1,20 +1,38 @@
-/proc/random_underwear(var/gender)
+proc/random_underwear(gender, species = "Human")
```
Comment (tigercat2000) [inline code/__HELPERS/mobs.dm]: These should have the absolute pathing `/proc/`

Comment (tigercat2000) [inline code/__HELPERS/mobs.dm]: well fuck a bit late

**Pull request merged**